File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022s/IZ08IO1/Bridge/src/Shape.java

import java.awt.Graphics;

public abstract class Shape
{
   protected int[] params;
   public Shape(String text,int count)
   {
      params=parameters(text);
      if(params.length!=count) throw new IllegalArgumentException("Wrong parameters count");
   }
   public abstract void Draw(Graphics g);
   public abstract String toXml();
   public static int[] parameters(String text)
   {
      String[] items=text.split("[\\s\t,;]+");
      int size=items.length;
      int[] params=new int[size];
      for(int i=0;i<size;++i) params[i]=Integer.parseInt(items[i]);
      return params;
   }
}

File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO2/BridgePatern/BridgePatern/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace BridgePatern
{
    public abstract class Shape
    {
        public static void ParseParams(string text, params Action<float>[] tb)
        {
            int size = tb.Length;
            string[] cells = text.Split("\t ;".ToArray(), StringSplitOptions.RemoveEmptyEntries);
            if (size != cells.Length) throw new Exception("Invalid parameters count");
            for (int i = 0; i < size; ++i) tb[i](float.Parse(cells[i]));
        }
        public abstract void Draw(Graphics g);
        public abstract string ToXml();
    }
}
